torch/_inductor/codegen/common.py

```diff
@@ -1019,6 +1019,9 @@ def indirect_indexing(var, size, check=True):
                             new_bounds = new_bounds | pos
```

Comment:
these three lines should be moved intto the if below and we should also update the bounds there manually